Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -168,18 +168,17 @@ static MediaKeyStatus mediaKeyStatusFromOpenCDM(std::string& keyStatus)
     return MediaKeyStatus::InternalError;
 }
 
-static size_t messageLength(std::string& message, std::string& request)
+static RefPtr<SharedBuffer> cleanResponseMessage(std::string& message)
 {
-    size_t length = 0;
-    std::string delimiter = ":Type:";
-    std::string requestType = message.substr(0, message.find(delimiter));
-    // FIXME: What is with this weirdness?
-    if (requestType.size() && requestType.size() == (request.size() + 1))
-        length = requestType.size() + delimiter.size();
-    else
-        length = request.length();
-    GST_TRACE("delimiter.size = %u, delimiter = %s, requestType.size = %u, requestType = %s, request.size = %u, request = %s", delimiter.size(), delimiter.c_str(), requestType.size(), requestType.c_str(), request.size(), request.c_str());
-    return length;
+    // If message does not begin with "message:", we bail out empty.
+    if (message.compare(0, 8, "message:"))
+        return { };
+
+    // Check the following 5 characters, if they are Type: we skip them as well.
+    unsigned offset = !message.compare(8, 5, "Type:") ? 13 : 8;
+
+    RefPtr<SharedBuffer> cleanResponseMessage = SharedBuffer::create(message.c_str() + offset, message.size() - offset);
```

Comment:
I don't dislike extra names that help the programmer understand the purpose (since the compiler elides them anyway) but following review comments I've received in the past, it seems the prevailing style is to just `return SharedBuffer::create` instead.